### PR TITLE
test(gnft): improve test coverage with edge cases and validations

### PR DESCRIPTION
**Pull request by junghoon-vans**
## Summary
- Add comprehensive halt state tests for all 7 operations (Mint, Burn, SafeTransferFrom, TransferFrom, Approve, SetApprovalForAll, SetTokenURI)
- Add unauthorized caller test for Mint function
- Add complete test coverage for SetApprovalForAll function
- Add invalid address test to TestApprove
- Enhance TestBalanceOf with multiple scenarios including zero balance, single NFT, multiple NFTs, and balance changes after transfers

## Test Coverage Improvements
- Halt state validation for all critical operations
- Access control validation (owner and staker checks)
- Edge cases for invalid addresses
- Balance tracking across different scenarios
- Approval operations (single token and operator approvals)

## Changes
- Added TestHaltedOperations covering all 7 main functions
- Extended TestMint with unauthorized caller case
- Created TestSetApprovalForAll with approve, revoke, and invalid operator tests
- Enhanced TestApprove with invalid address case
- Improved TestBalanceOf with 5 different scenarios
- Added helper functions in testutils.gno (setHalted, getAdminAddress)

### CODE REVIEW

File: contract/r/gnoswap/gnft/testutils.gno
Comment (jinoosss) [inline contract/r/gnoswap/gnft/testutils.gno]: Please rename the file to `_helper_test.gno`.

File: contract/r/gnoswap/gnft/testutils.gno [hunk 1]
```diff
@@ -4,6 +4,9 @@ import (
 	"testing"
 
 	"gno.land/p/demo/tokens/grc721"
+	"gno.land/p/gnoswap/rbac"
+	"gno.land/r/gnoswap/access"
+	"gno.land/r/gnoswap/halt"
 )
 
 func InitGNFTTest(t *testing.T) {
```
Comment (jinoosss) [inline contract/r/gnoswap/gnft/testutils.gno]: remove this function.
